File: src/test/java/helper/GifHelper.java
package helper;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.imageio.stream.FileImageOutputStream;
import javax.imageio.stream.ImageOutputStream;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import java.util.List;

public class GifHelper {

    private static final Logger log = LoggerFactory.getLogger(GifHelper.class);

    public static void createGif(File outputFile, List<BufferedImage> images) {
        createGif(outputFile, images, 1, true);
    }

    public static void createGif(File outputFile, List<BufferedImage> images, int timeBetweenFramesMS, boolean loopContinuously) {
        int imageType = images.iterator().next().getType();

        try (ImageOutputStream output = new FileImageOutputStream(outputFile);
             GifSequenceWriter writer = new GifSequenceWriter(output, imageType, timeBetweenFramesMS, loopContinuously)) {
            images.forEach(writer::writeToSequence);
        } catch (IOException e) {
            log.error("An error occurred whilst creating gif file '" + outputFile.getName() + "': ", e);
        }
    }
}

File: src/test/kotlin/net/scriptgate/helper/GifHelper.kt
package net.scriptgate.helper

import org.slf4j.LoggerFactory
import java.awt.image.BufferedImage
import java.io.File
import java.io.IOException
import java.util.function.Consumer
import javax.imageio.stream.FileImageOutputStream

object GifHelper {

    private val log = LoggerFactory.getLogger(GifHelper::class.java)

    @JvmOverloads
    fun createGif(outputFile: File, images: List<BufferedImage>, timeBetweenFramesMS: Int = 1, loopContinuously: Boolean = true) {
        val imageType = images.iterator().next().type

        try {
            FileImageOutputStream(outputFile).use { output -> GifSequenceWriter(output, imageType, timeBetweenFramesMS, loopContinuously).use { writer -> images.forEach(Consumer<BufferedImage> { writer.writeToSequence(it) }) } }
        } catch (e: IOException) {
            log.error("An error occurred whilst creating gif file '" + outputFile.name + "': ", e)
        }

    }
}
